client/test/e2e.test.js

```diff
@@ -0,0 +1,112 @@
+import test from 'tape';
+import puppeteer from 'puppeteer';
+
+function sleep(ms) {
+    return new Promise(resolve => setTimeout(resolve, ms));
+}
+
+const width = 1920;
+const height = 1080;
+
+function browser(width, height) {
+    return puppeteer.launch({
+    headless: true,
+    args: [`--window-size=${width},${height}`],
+    // slowMo: 20 // For testing
+    });
+}
+
+test('e2e-client-test-valid-actions', async (t) => {
```

Comment:
The name of the test can and should be human readable. Can you use a descriptive natural language phrase here?

It's also worth considering whether to meld all the assertions into a single test. I'm not sure "valid" vs. "invalid" is a super useful distinction for client testing.

Reply:
Fixed in 5b3ea58